src/main/java/com/achdev/onlinebookstoreapp/service/impl/UserServiceImpl.java

```diff
@@ -23,6 +27,17 @@ public UserDto register(UserRegistrationRequestDto requestDto) throws Registrati
             throw new RegistrationException("User with email: " + email + " already exists");
         }
         User user = userMapper.toModel(requestDto);
+        setRoleIdsForUser(user);
```

Comment:
example, add throw an exception
```suggestion
        user.setRoles(Set.of(roleRepository.findByName(RoleName.USER)));
```

Reply:
Ok